Objective-C/Internal/Replicator/CBLWebSocket.mm

```diff
@@ -374,10 +380,23 @@ - (void) _sendWebSocketRequest {
     // Construct the HTTP request:
     for (Dict::iterator header(_options[kC4ReplicatorOptionExtraHeaders].asDict()); header; ++header)
         _logic[slice2string(header.keyString())] = slice2string(header.value().asString());
-    slice cookies = _options[kC4ReplicatorOptionCookies].asString();
-    if (cookies)
-        [_logic addValue: cookies.asNSString() forHTTPHeaderField: @"Cookie"];
-
+    
+    slice sessionCookie = _options[kC4ReplicatorOptionCookies].asString();
+    if (sessionCookie.buf)
+        [_logic addValue: sessionCookie.asNSString()  forHTTPHeaderField: @"Cookie"];
+    
+    C4Error err = {};
+    C4Address addr = {};
+    [_remoteURL c4Address: &addr];
+    C4SliceResult cookies = c4db_getCookies(_c4db, addr, &err);
```

Comment:
I don't have the info about its thread safe behavior. Can you check if we need to lock the db or not?

Reply:
What do you think about doing is in CBLReplicator versus here?

Reply:
Look lie we save the cookie here so getting the cookies here seems to be inline.